source/core/NumericString.cpp

```diff
@@ -1291,6 +1293,38 @@ IntMax ScanIntValues(char formatChar, char* beginPointer, char** endPointer)
     return intValue;
 }
 
+char validTrueValues[5][10] = { "y", "t", "true", "on", "yes" };
+char validFalseValues[5][10] = { "n", "f", "false", "off", "no" };
+
+Boolean ScanBooleanValue(char* beginPointer, char** endPointer)
+{
+    SubString s(beginPointer);
+    *endPointer = beginPointer;
+
+    for (const char* trueValue : validTrueValues) {
+        if (s.CompareCStrIgnoreCase(trueValue)) {
+            *endPointer += strlen(trueValue);
+            return true;
+        }
+    }
+
+    for (const char* falseValue : validFalseValues) {
+        if (s.CompareCStrIgnoreCase(falseValue)) {
+            *endPointer += strlen(falseValue);
+            return false;
+        }
+    }
+
+    return false;
```

Comment:
Is this how CG and NG behave, returning false when none of the "true", "false" values are found? Do we have a test for this?

Reply:
I tested it and it is based on the default value wired in, so what I have here is incorrect. I should be returning the default value.